### PR TITLE
[SPARK-40391][SQL][TESTS] Test the error class UNSUPPORTED_FEATURE.JDBC_TRANSACTION

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to Add one test for the error class UNSUPPORTED_FEATURE.JDBC_TRANSACTION to QueryExecutionErrorsSuite.

### Why are the changes needed?
Add one test for the error class UNSUPPORTED_FEATURE.JDBC_TRANSACTION to QueryExecutionErrorsSuite.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Manual test：

```
./build/sbt "sql/testOnly *QueryExecutionErrorsSuite*"
```

All tests passed.

### CODE REVIEW

File: sql/core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (panbingkun) [inline sql/core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: https://www.baeldung.com/mockito-final#configure-mocktio
Configure Mockito for Final Methods and Classes
image

**Comment by panbingkun** [comment]
cc @MaxGekk

**Comment by MaxGekk** [comment]
@panbingkun Could you fix scala style failure:
```
[error] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:34:0: org.apache.spark.sql.execution.datasources.jdbc. is in wrong order relative to 
```

**Comment by panbingkun** [comment]
> @panbingkun Could you fix scala style failure:
> 
> ```
> [error] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:34:0: org.apache.spark.sql.execution.datasources.jdbc. is in wrong order relative to 
> ```

Done.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**